classes/Leth/IPAddress/IP/Address.php

```diff
@@ -26,33 +28,34 @@
  */
 abstract class Address implements \ArrayAccess
 {
-	const IP_VERSION = -1;
-	const FORMAT_FULL = 0;
-	const FORMAT_COMPACT = 1;
+	public const IP_VERSION  = -1;
+	public const FORMAT_FULL    = 0;
+	public const FORMAT_COMPACT = 1;
 
 	/**
 	 * Internal representation of the address. Format may vary.
-	 * @var mixed
+	 * @var numeric
 	 */
 	protected $address;
 
-	/**
-	 * Create an IP address object from the supplied address.
-	 *
-	 * @param string $address The address to represent.
-	 * @return IP\Address An instance of a subclass of IP\Address; either IPv4\Address or IPv6\Address
-	 */
-	public static function factory($address)
+    /**
+     * Create an IP address object from the supplied address.
+     *
+     * @param IP\Address|int|string|\Math_BigInteger $address The address to represent.
+     *
+     * @return \Leth\IPAddress\IP\Address|\Leth\IPAddress\IPv4\Address|\Leth\IPAddress\IPv6\Address An instance of a subclass of IP\Address; either IPv4\Address or IPv6\Address
+     */
+	public static function factory(IP\Address|int|string|\Math_BigInteger $address): IP\Address|Ipv4\Address|IPv6\Address
```

Comment:
Misaligned indentation. Sorry, I legally have to point out one formatting thing during a code review.

Reply:
For all code I'd like to use [PHPFixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer).
What do You think about it?

Reply:
I can't comment on PHPFixer itself, but I'd strongly encourage adopting automatic tooling to enforce code style; it saves everyone time, avoids a lot of nitpicky feedback which is not fun for anyone, and usually makes code easier to read and interpret correctly!

Reply:
Yeah same, but I think that warrants its own PR.